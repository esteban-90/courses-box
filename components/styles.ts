import { css, SerializedStyles } from '@emotion/react'
import { AppTheme } from '@/styles/themes'

export type Color = 'primary' | 'secondary' | 'danger' | 'warning'

export const boxShadow = (shadowColor1: string, shadowColor2: string, inset = false): SerializedStyles => {
  const insetStr = inset ? 'inset' : false

  return css`
    box-shadow: 0.5vmin 0.5vmin 1vmin ${shadowColor1} ${insetStr}, -0.5vmin -0.5vmin 1vmin ${shadowColor2} ${insetStr};
  `
}

export const transition = (): SerializedStyles => {
  return css`
    transition: all 0.4s ease;
  `
}

export const getColors = (theme: AppTheme, color?: Color): SerializedStyles => {
  switch (color) {
    case 'secondary':
      return css`
        color: ${theme.font.regular};
      `

    case 'primary':
    case 'danger':
      return css`
        background: ${theme.components[color]};
        color: ${theme.font.button};
      `

    case 'warning':
      return css`
        background: ${theme.components[color]};
        color: ${theme.font.warning};
      `

    default:
      return css``
  }
}
