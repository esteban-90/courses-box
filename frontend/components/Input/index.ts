export * from './Input'
export * from './Feedback'
