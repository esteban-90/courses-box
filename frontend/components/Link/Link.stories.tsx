import { ComponentStoryObj, ComponentMeta } from '@storybook/react'
import { expect } from '@storybook/jest'
import { screen } from '@storybook/testing-library'
import { Link } from './Link'

export default {
  title: 'Content/Link',
  component: Link,
} as ComponentMeta<typeof Link>

export const BasicLink: ComponentStoryObj<typeof Link> = {
  async play() {
    await expect(screen.getByRole('link')).toBeInTheDocument()
  },

  args: {
    children: 'Hands-On React. Build advanced React JS Frontend with expert',
    href: '/hands-on-reactjs',
  },
}
