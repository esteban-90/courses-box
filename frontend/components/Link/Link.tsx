import styled from '@emotion/styled'
import { LinkProps } from '@/types'

export const Link = styled.a<LinkProps>`
  all: unset;
  cursor: pointer;
  color: ${({ theme }) => theme.font.regular};
  text-decoration: ${({ underline }) => (underline ? 'underline' : 'none')};

  &:hover {
    opacity: 0.7;
  }
`
