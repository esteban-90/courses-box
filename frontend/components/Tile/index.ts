export * from './Tile'
export * from './CenteredTile'
