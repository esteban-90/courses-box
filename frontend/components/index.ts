export * from './Button'
export * from './Checkbox'
export * from './Course'
export * from './Icon'
export * from './IconButton'
export * from './Input'
export * from './Layout'
export * from './Link'
export * from './Logo'
export * from './Switch'
export * from './Tile'
