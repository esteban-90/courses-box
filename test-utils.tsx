import { FC, ReactElement, ReactNode } from 'react'
import { render, RenderOptions, RenderResult } from '@testing-library/react'
import { ThemeProvider } from '@emotion/react'
import { Themes } from '@/styles/themes'

const Wrapper: FC<{ children: ReactNode }> = ({ children }): JSX.Element => {
  return <ThemeProvider theme={Themes.light}>{children}</ThemeProvider>
}

const customRender = (ui: ReactElement, options?: RenderOptions): RenderResult => {
  return render(ui, { wrapper: Wrapper, ...options })
}

export * from '@testing-library/react'
export { customRender as render }

// 14
