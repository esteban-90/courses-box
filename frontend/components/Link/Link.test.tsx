import { render } from '@/test-utils'
import { Link } from '@/components/Link'

describe('Link test cases', () => {
  it('Render check', () => {
    const element = <Link href='/hands-on-reactjs'>Hands-On React. Build advanced React JS Frontend with expert</Link>
    const { asFragment } = render(element)

    expect(asFragment()).toMatchSnapshot()
  })
})
