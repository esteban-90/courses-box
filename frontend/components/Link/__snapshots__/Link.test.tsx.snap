// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link test cases Render check 1`] = `
<DocumentFragment>
  <a
    class="css-1nz5qw1"
    href="/hands-on-reactjs"
  >
    Hands-On React. Build advanced React JS Frontend with expert
  </a>
</DocumentFragment>
`;
